### PR TITLE
Refactor youtube import

### PR DESCRIPTION
**Pull request by perryr16**
* created a key generation method 
* value generation method
* refactored hash generation method using helpers

### CODE REVIEW

File: app/controllers/admin/tutorials_controller.rb [hunk 1]
```diff
@@ -22,9 +22,7 @@ def new
 
   def update
     tutorial = Tutorial.find(params[:id])
-    if tutorial.update(tutorial_params)
-      flash[:success] = "#{tutorial.title} tagged!"
-    end
+    flash[:success] = "#{tutorial.title} tagged!" if tutorial.update(tutorial_params)
     redirect_to edit_admin_tutorial_path(tutorial)
```
Comment (perryr16) [inline app/controllers/admin/tutorials_controller.rb:26]: rubocop -a

File: app/models/tutorial.rb [hunk 2]
```diff
@@ -14,16 +14,20 @@ def playlist_video_json
...
-      video_params = {}
-      video_params[:title] = video[:snippet][:title]
-      video_params[:description] = video[:snippet][:description]
-      video_params[:video_id] = video[:contentDetails][:videoId]
-      video_params[:thumbnail] = video[:snippet][:thumbnails][:default][:url]
-      video_params[:position] = video[:snippet][:position]
-      playlist_video_params << video_params
+    playlist_video_json[:items].map do |video|
+      Hash[playlist_video_params_keys.zip(playlist_video_params_values(video))]
     end
```
Comment (perryr16) [inline app/models/tutorial.rb:19]: given keys and values, zip together on each iteration to create an array of video_param hashes

**Review by perryr16 — COMMENTED** [review]
youtube playlist video params refactor

File: app/models/tutorial.rb [hunk 3]
```diff
@@ -14,16 +14,20 @@ def playlist_video_json
...
+  def playlist_video_params_values(video)
+    [video[:snippet][:title],
+     video[:snippet][:description],
+     video[:contentDetails][:videoId],
+     video[:snippet][:thumbnails][:default][:url],
+     video[:snippet][:position]]
+  end
+
+  def playlist_video_params_keys
+    %i[title description video_id thumbnail position]
```
Comment (perryr16) [inline app/models/tutorial.rb:31]: `rubocop -a`did this. it is equivalent to `[:title, :description...]`

File: .travis.yml [hunk 4]
```diff
@@ -18,3 +18,4 @@ deploy:
     secure: nXSlNk6ytMdtTaSywTK6HP3btTtPHjNkuujh2VU/JAeI2S3BKKLCTtmgsWYKbWFL0CSt8LCqWGp2kTDLozMgYfuoGP1fqSM+F2p/c1C/pH+bHa7QPu/qp09yaQc9IF4sDKPbimXCbcioU2JDeURMWRPc0WJoQGCpu5ynFMvvU+FWJXpcckL1DrEJbPcsMgvYadIzasF9Vl0QKnLs5wDFoJGPN2YZ0A+Za8aKoLJRoaITyAcIXA8HFE+2YMTBU8CcDiJQyk83b1Glvx9i3e5ZegJVeOzrpR0osCnz6y9613oNuSj2ViyPnLm7Y/WJMtd793scl2IDJtpjAabASGFydwU4AS3u6/dfa7cVykDb991IV2YBguU/vf4qVqGCh/K/KiKbO5fC2T/4fmNbQ/X2JiRO5m8O3RYvq4RIkTh72pjHvyfJGXl/PN+xb1Xl03cmDcrqC0LnNISl8FMmOO7tbRo7DA5c0X0HUnkSAhMXN3Yt/dLCCPIVCWbr2jeYxynDA34B8FimmGrfqTSmyqMqtILW7sAlMKxQfX+19otn+zFY3i/+RtQchFJhoMlcRW1WiAJT8xfkItqShaB0d5hhpxe5kC+hpohfSZirrKnQwwOXlC3sn0bJWq0ibBrogIOD1YxjgpHYI+2dWOmugpGAtEf9wVYPC2hhByKxYSatYys=
   app: brownfield-of-dreams-sb-rp
   run: rails db:migrate
+  GITHUB_API_KEY: "token 484cabdee3b4e5de4226ed80470600b274e435d9"
```
Comment (perryr16) [inline .travis.yml]: I think I need to remove this from the yml file

File: app/models/tutorial.rb (hunk 2 above)
Comment (stellakunzang) [reply app/models/tutorial.rb:19]: nice! gotta love that zip!

**Pull request merged**